verifier/src/lib.rs

```diff
@@ -70,25 +70,55 @@ pub use errors::VerifierError;
 // ================================================================================================
 /// Verifies that the specified computation was executed correctly against the specified inputs.
 ///
-/// Specifically, for a computation specified by `AIR` and `HashFn` type parameter, verifies that the provided
-/// `proof` attests to the correct execution of the computation against public inputs specified
-/// by `pub_inputs`. If the verification is successful, `Ok(())` is returned.
+/// Specifically, for a computation specified by `AIR` and `HashFn` type parameter, verifies that 
+/// the provided `proof` attests to the correct execution of the computation against public inputs
+/// specified by `pub_inputs`. If the verification is successful, `Ok(())` is returned.
 ///
 /// # Errors
 /// Returns an error if combination of the provided proof and public inputs does not attest to
 /// a correct execution of the computation. This could happen for many various reasons, including:
 /// - The specified proof was generated for a different computation.
 /// - The specified proof was generated for this computation but for different public inputs.
+/// - The specified proof was generated with parameters not providing an acceptable security level.
 #[rustfmt::skip]
 pub fn verify<AIR, HashFn, RandCoin>(
     proof: StarkProof,
     pub_inputs: AIR::PublicInputs,
+        acceptable_options: &AcceptableOptions,
 ) -> Result<(), VerifierError> 
 where 
     AIR: Air, 
     HashFn: ElementHasher<BaseField = AIR::BaseField>,
     RandCoin: RandomCoin<BaseField = AIR::BaseField, Hasher = HashFn>,
 {
+    // check that `proof` was generated with an acceptable set of parameters from the point of view
+    // of the verifier
+    match acceptable_options {
+            AcceptableOptions::MinConjecturedSecurity(minimal_security) => {
+                let proof_security = proof.security_level::<HashFn>(true);
+                if proof_security < *minimal_security {
+                    return Err(VerifierError::InsufficientConjecturedSecurity(
+                        *minimal_security,
+                        proof_security,
+                    ));
+                }
+            }
+            AcceptableOptions::MinProvenSecurity(minimal_security) => {
+                let proof_security = proof.security_level::<HashFn>(false);
+                if proof_security < *minimal_security {
+                    return Err(VerifierError::InsufficientProvenSecurity(
+                        *minimal_security,
+                        proof_security,
+                    ));
+                }
+            }
+            AcceptableOptions::OptionSet(options) => {
+                if !options.iter().any(|opt| opt == proof.options()) {
+                    return Err(VerifierError::UnacceptableProofOptions);
+                }
+            }
+        }
```

Comment:
I would probably move this logic into a function implemented on `AcceptableOptions`. Maybe something like:

```Rust
impl AcceptableOptions {
    pub fn validate(&self, proof: &StarkProof) -> Result<(), VerifierError> {
        ...
    }
}
```